lib/ipam/ipam.go

```diff
@@ -299,10 +299,25 @@ func (c ipamClient) autoAssign(ctx context.Context, num int, handleID *string, a
 		logCtx = logCtx.WithField("handle", *handleID)
 	}
 	logCtx.Info("Looking up existing affinities for host")
-	affBlocks, err := c.blockReaderWriter.getAffineBlocks(ctx, host, version, pools)
+	affBlocks, affBlocksToRelease, err := c.blockReaderWriter.getAffineBlocks(ctx, host, version, pools)
```

Comment:
I think we need to check the IP pool here to make sure the `affBlockToRelease` isn't in a pool that selects this node. 

Take this example:
- There are two pools which select this node - `pool1`, `pool2`
- There are two pools which do _not_ select this node - `pool3`, `pool4`
- I pass in an explicit pool which doesn't select this node - `pool3`.

We don't want to release any blocks that are affine to this node that are within `pool1` or `pool2`. But, we do want to release any empty blocks within `pool4`.

Reply:
Might need to augment `determinePools` or something to return both the pools to use for the allocation, as well as all pools, so that we can match against other pools which may not have been provided on this particular request.